extra_geom/detectors.py

```diff
@@ -768,6 +768,81 @@ def offset(self, shift, *, modules=np.s_[:], tiles=np.s_[:]):
         ])
 
 
+class CrystFEL_Geometry(DetectorGeometryBase):
+    """A generic detector layout based on the CrystFEL geom file.
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+    """
+    detector_type_name = 'Generic Detector'
+    pixel_size = None   # should be in meters
+    frag_ss_pixels = None
+    frag_fs_pixels = None
+    expected_data_shape = ()
+    n_quads = 0
+    n_modules = None
+    n_tiles_per_module = None
+
+    def __init__(self, pixel_size: float, fast_pixels: int, slow_pixels: int,
+                 corner_coordinates: [np.ndarray] = [np.zeros(3)],
+                 fs_vec: np.ndarray = np.array([0, 1, 0]),
+                 ss_vec: np.ndarray = np.array([1, 0, 0]),
+                 n_tiles_per_module: int = 1,
+                 tile_offset: float = None,
+                 tile_vec: np.ndarray = None,
+                 ):
```

Comment:
I'm a bit unsure about this, but if we're aiming for a more *generic* class rather than a *simple* one, we might allow this input with a classmethod like `from_simple_description`, e.g.

```python
geom = GenericDetector.from_simple_description(
    pixel_size=2e-4,
    ...
)
```

This would allow us to have other ways to construct the class if we one day needed more flexibility (e.g. modules arranged in different orientations, like CSPAD).

Reply:
I tend to agree, but would we have a _default_ method of construction then?

Reply:
The current classes don't really have a useful *default* constructor (all the examples use things like `AGIPD_1MGeometry.from_crystfel_geometry()` or `LPD_1MGeometry.from_quad_positions()` rather than `LPD_1MGeometry()`). I don't think that's a big problem.

Reply:
Done, docs updated accordingly.

Reply:
Thanks :)